src/components/GlMap.svelte

```diff
@@ -194,6 +167,31 @@
       map.remove();
     }
   });
+
+  // We check map and mapViewProps here to ensure this reacts to changes to
+  // either
+  $: map && updateMapFromProps(mapViewProps);
+
+  $: map && updateMapStyle(url, style);
```

Comment:
Is there a good reason not to also include `map` as a parameter to these functions? Not sure I'm understanding where we draw the line here.

Reply:
Just updated to add `map` as an arg which makes sense and is more clear with this pattern. Initially I thought it felt a little excessive to pass it in just to return if it wasn't valid, but both functions do actually use `map` so this makes sense.

The other reactive blocks like `$: map && (map.showCollisionBoxes = showCollisions);` feel ok to leave as is to me without a function, but let me know if you disagree and I can toss these in a function as well.